packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -53,7 +61,81 @@ export function TelemetryView(): React.ReactElement {
 		});
 	}, [setTelemetryEvents]);
 
-	return <_TelemetryView telemetryEvents={telemetryEvents}></_TelemetryView>;
+	if (telemetryEvents.length > pageLimit) {
+		setTelemetryEvents(telemetryEvents.slice(0, pageLimit));
+	}
+
+	console.log(`Telemetry Events length: ${telemetryEvents.length}`);
+
+	return (
+		<Stack>
+			<StackItem>
+				<_PagationSelection
+					limit={pageLimit}
+					onChangeSelection={(key): void => setPageLimit(key)}
+				/>
+			</StackItem>
+			<StackItem>
+				<_TelemetryView telemetryEvents={telemetryEvents}></_TelemetryView>
+			</StackItem>
+		</Stack>
+	);
+}
+
+/**
+ * {@link _PagationSelectionProps} input props.
+ *
+ * @internal
+ */
+export interface _PagationSelectionProps {
+	/**
+	 * The current page limit.
+	 * @defaultValue 100
```

Comment:
If we extract the default limit into a constant, we can link to it from here.

Reply:
Updated